src/GraphQL/Execution/ExecutionContext.cs

```diff
@@ -62,20 +63,27 @@ public class ExecutionContext : IExecutionContext
 
         private readonly List<Array> _trackedArrays = new List<Array>();
 
-        internal void TrackArray(Array array)
-        {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
+        TElement[] IExecutionArrayPool.Rent<TElement>(int minimumLength) => RentSharedArray<TElement>(minimumLength);
 
+        /// <inheritdoc cref="IExecutionArrayPool.Rent{TElement}(int)"/>
+        public TElement[] RentSharedArray<TElement>(int minimumLength)
+        {
+            var array = ArrayPool<TElement>.Shared.Rent(minimumLength);
             lock (_trackedArrays)
                 _trackedArrays.Add(array);
+            return array;
         }
 
-        internal void ReturnArrays()
+        /// <summary>
+        /// Releases any rented arrays back to the backing memory pool.
+        /// </summary>
+        public void Dispose()
         {
             // lock is not required because at this time work with ExecutionContext has already been completed
             foreach (var array in _trackedArrays)
                 array.Return();
+
+            _trackedArrays.Clear();
```

Comment:
At first I also wrote that, but therefore I thought that it does not make sense, because `ExecutionContext` contains references to a huge variety of other information, see properties. Either all or nothing needs to be cleared. I decided not to clear anything.